src/rsp_restspawner/spawner.py

```diff
@@ -95,114 +167,291 @@ def _client(self) -> AsyncClient:
+
+        Raises
+        ------
+        httpx.HTTPError
+            Raised on failure to talk to the lab controller or a failure
+            response from the lab controller.
+        """
         r = await self._client.delete(
             self._controller_url("labs", self.user.name),
             timeout=300.0,
             headers=self._admin_authorization(),
         )
-        if r.status_code == 202 or r.status_code == 404:
-            # We're deleting it, or it wasn't there to start with.
+        if r.status_code == 404:
+            # Nothing to delete, treat that as success.
             return
-        raise SpawnerError(r)
+        else:
+            r.raise_for_status()
 
     async def poll(self) -> Optional[int]:
-        """
-        Check if the pod is running.
+        """Check if the pod is running.
 
-        If it is, return None.  If it has exited, return the return code
-        if we know it, or 0 if it exited but we don't know how.
-
-        Because we do not have direct access to the pod's exit code, we
-        are here going to return 0 for "The pod does not exist from the
-        perspective of the lab controller" (which assumes a good or unknown
-        exit status) and 1 for "We tried to start a pod, but it failed," which
-        implies a failure (i.e. non-zero) exit status.
+        Returns
+        -------
+        int or None
+            If the pod is starting, running, or terminating, return `None`.
+            If the pod does not exist, return 0. If the pod exists in a failed
+            state, return 1.
+
+        Notes
+        -----
+        In theory, this is supposed to be the exit status of the Jupyter lab
+        process. This isn't something we know in the classic sense since the
+        lab is a Kubernetes pod. We only know that something failed if the
+        record of the lab is hanging around in a failed state, so use a simple
+        non-zero exit status for that. Otherwise, we have no way to
+        distinguish between a pod that was shut down without error and a pod
+        that was stopped, so use an exit status of 0 in both cases.
         """
         r = await self._client.get(
             self._controller_url("labs", self.user.name),
             headers=self._admin_authorization(),
         )
         if r.status_code == 404:
             return 0  # No lab for user.
-        if r.status_code != 200:
-            raise SpawnerError(r)
+        else:
+            r.raise_for_status()
         result = r.json()
         if result["status"] == LabStatus.FAILED:
             return 1
         else:
             return None
 
     async def options_form(self, spawner: Spawner) -> str:
+        """Retrieve the options form for this user from the lab controller.
+
+        Parameters
+        ----------
+        spawner
+            Another copy of the spawner (not used). It's not clear why
+            JupyterHub passes this into this method.
```

Comment:
I once convinced myself it was something to do with functools and partial application, but I wouldn't swear I was right.